### PR TITLE
[PRODDEV-414] New Admin for Virtual Y Homepage

### PR DESCRIPTION
**Pull request by Sardis93**
**Related Issue/Ticket:**

[PRODDEV-414](https://openy.atlassian.net/browse/PRODDEV-414)

## Steps to test:

- [ ] Visit Virtual Y settings page, accessible via toolbar "**Virtual Y**" -> "**Virtual YMCA settings**".
- [ ] Confirm **Switch to Legacy View** checkbox replaces **Components Settings** with **Legacy View** fieldset.
- [ ] Confirm that any field for new components (within **Components Settings**) can be updated correctly.

## Quality checks:

Please check these boxes to confirm this PR covers the following cases:

- Maintaining our upgrade path is essential. Check one or the other:
  - [x] This PR  provides updates via `hook_update_N` or other means.
  - [ ] No updates are necessary for this change.
- Front end fixes should be tested against all of the Open Y Themes.
  - [ ] Tested against Carnation
  - [ ] Tested against Lily
  - [ ] Tested against Rose
  - [x] This change does not contain front-end fixes.
- [x] I have flagged this PR "Needs Review" or pinged the VY devs/QA
  team in Slack

### CODE REVIEW

**Review by anpolimus — APPROVED** [review]

**Review by anpolimus — APPROVED** [review]

File: src/Form/GCSettingsForm.php [hunk 1]
```diff
@@ -80,7 +81,108 @@ public function buildForm(array $form, FormStateInterface $form_state): array {
       '#required' => TRUE,
     ];
 
-    $form['app_settings']['components'] = [
+    $form['app_settings']['switch_legacy_view'] = [
+      '#title' => $this->t('Switch to Legacy View'),
+      '#description' => $this->t('Legacy View has the following components: Virtual Y Video, Live streams, Virtual meetings, Blog posts.'),
```
Comment (hamrant) [inline src/Form/GCSettingsForm.php:86]: Legacy View has the following components `on the Home page`: Virtual Y Video, Live streams, Virtual meetings, Blog posts.

File: src/Form/GCSettingsForm.php [hunk 2]
```diff
@@ -80,7 +81,108 @@ public function buildForm(array $form, FormStateInterface $form_state): array {
...
+      '#type' => 'checkbox',
+      '#default_value' => $config->get('switch_legacy_view') ?? FALSE,
+    ];
+
+    $this->prepareComponents($form, $config);
+    $this->prepareLegacyView($form, $config);
```
Comment (hamrant) [inline src/Form/GCSettingsForm.php:92]:
```php
    $this->prepareComponents($form, $config);
    $this->prepareLegacyView($form, $config);
```

These functions are very similar. I think we can simplify this and avoid copy/past. Let's add an additional argument `array $components` and maybe container_id and merge to one function

**Review by hamrant — CHANGES_REQUESTED** [review]

File: src/Form/GCSettingsForm.php (hunk 2 above)
Comment (hamrant) [reply src/Form/GCSettingsForm.php]: We with @Sardis93 decided to rollback to the previous version, so let's resolve this conversation

**Review by hamrant — APPROVED** [review]

**Comment by AnastasiiaPys** [comment]
retest

**Comment by AnastasiiaPys** [comment]
Tested on macOS(Chrome).
Works as expected.

**Pull request merged**